src/app/projects/project/AssignmentList.js

```diff
@@ -22,8 +22,20 @@ export default function Assignments({ personId, projectId }) {
   const { t } = useTranslation();
   const { data: project } = useProject(projectId);
   const { docs: allAssignments } = useMySlots(personId);
-  const assignments = Object.entries(
-    filterObj(allAssignments, ([id, doc]) => doc.projectId === projectId),
+
+  const currentTimestamp = new Date(); // current time
+
+  let assignments = Object.entries(
+    filterObj(
+      allAssignments,
+      ([id, doc]) =>
+        doc.projectId === projectId && new Date(doc.ends) >= currentTimestamp,
+    ),
+  );
+
+  // sort by date
+  assignments = assignments.sort(
```

Comment:
Array's `sort()` function modifies the array, so you don't have to re-declare the variable. So you can simply:
```js
assignments.sort(
```

Then you can change `assignments` variable to be `const` and not `let`.